### DISCUSSION TITLE
Composer Installs wrong version by default

### DISCUSSION THREAD

**Discussion by mhenkel1** [spatie/pdf-to-image#178] [Q&A] [answered]
Hey, just wanted to let you know, that if you try to install your package with composer like you described in the readme, it will install branch v2.

composer require spatie/pdf-to-image

Results in this being installed: https://packagist.org/packages/spatie/pdf-to-image#v2.x-dev

Just fyi.
Awesome package by the way, thumbs up!

**Comment by pryley** [comment, marked as answer]
I had this issue also, `composer require spatie/pdf-to-image` was installing `v2.x-dev 6718a89`.

When I changed the version in composer.json from `^2.1` to `2.1` and hit `composer update`, I got the following error message:

```
Problem 1
    - Root composer.json requires spatie/pdf-to-image 2.1 -> satisfiable by spatie/pdf-to-image[2.1.0].
    - spatie/pdf-to-image 2.1.0 requires ext-imagick * -> it is missing from your system. Install or enable PHP's imagick extension.
```

So because I didn't have the imagick extension enabled in my PHP, composer was installing the latest version of v2 that it could without the "ext-imagick" composer requirement (i.e. v2.x.dev)

I checked PHP by running:

```bash
php -m | grep imagick
```

Sure enough there was no imagick module loaded. Fixing this was easy:

```bash
pecl install imagick; valet restart
```

Now `composer update` gives me the latest `v2.1`
